CDP4Composition/Mvvm/BrowserViewModelBase.cs

```diff
@@ -373,7 +373,18 @@ public bool HasUpdateStarted
         public IRowViewModelBase<Thing> SelectedThing
         {
             get { return this.selectedThing; }
-            set { this.RaiseAndSetIfChanged(ref this.selectedThing, value); }
+            set
+            {
+                if (value == null || value.ContainedRows.Count == 0)
+                {
```

Comment:
this logic should not be in the setter here.

Reply:
Remove the logic and checking directly under context menu creation